### ISSUE TITLE
Use case: Stream arrays from redis store

### ISSUE THREAD

**Issue by jhamman** [mrocklin/streamz#69]
We are developing a tool for streaming N-D arrays from low level languages (Fortran/C) to Python. We are using Redis and are in the process of designing an Xarray backend (https://github.com/nbren12/geostreams/issues/6) as the user facing API. We are exploring different stream handling intermediaries that will allow us do many of the common reactive programing tasks on a collection of key/value pairs mapped to Xarray objects. Ideally, we come up with a solution that works well with dask arrays too.

@nbren12 may want to expand my initial description.

References:
1. [Geostreams Issue](https://github.com/nbren12/geostreams/issues/6)
1. [Geostreams Wiki](https://github.com/nbren12/geostreams/wiki) describing project

cc @nbren12, @phargogh, @ajijohn, @mrocklin

**Comment by mrocklin** [comment]
It would be interesting to hear what API you would expect from a low-level streaming library that you would want to exist in order to support your project.  Do existing solutions work?  If not why not?

**Comment by mrocklin** [comment]
Note that the other current user group of this library handles image processing pipelines.  You might want to chat with people like @CJ-Wright @danielballan and @ordirules to get a sense of their experience so far.

**Comment by danielballan** [comment]
Yes, this is interesting. We'd be happy to chat.

**Comment by mrocklin** [comment]
> Yes, this is interesting. We'd be happy to chat.
￼￼
Did this ever happen?  If not, are people free today?  I'm in use-case collecting mode for streamz, so I would find this helpful.

**Comment by CJ-Wright** [comment]
I'm also available today.

**Comment by ordirules** [comment]
yep, should be able to make time. @mrocklin  can you email us all?

**Comment by mrocklin** [comment]
Any time constraints?  I'm busy after 4:30pm Eastern US.

**Comment by danielballan** [comment]
I too am available.

**Comment by ordirules** [comment]
None within your time constraints

**Comment by danielballan** [comment]
I have to start running for the last train by about 4:10 ET.

**Comment by jhamman** [comment]
I've bumped the time to 315 ET so @nbren12 and I can join. Hopefully that works for everyone.

**Comment by nbren12** [comment]
It was fun meeting everyone today. I hope we can continue to talk about some of these issues together. I just profiled the send/recv performance of redis. Checkout https://gist.github.com/d7283b77634f52d6c1afd62c67bfc254.  I get about 1GB/s of write performance, and 500 MB/s of read performance using `redis-py`. I am not sure if this is the best possible, because `redis-py` might be doing some allocations behind the scenes.

**Comment by nbren12** [comment]
I just added pyarrow. It's blazing fast as expected!

[gist](https://gist.github.com/nbren12/598cb54aa60be2b105ced15149558bb8)

**Comment by ordirules** [comment]
thanks for sharing!
